src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -132,6 +132,28 @@ public function findAvailableEpisodesByCategoryAncestry(
         return $this->resolveParents($result);
     }
 
+    public function findByCoreEntityMembership(int $entityId, ?string $groupType, ?int $limit, int $offset)
+    {
+        if ($groupType === null) {
```

Comment:
Can you do this the same way as the other stuff in the class, make $groupType default to 'Group' and validate what's passed in against the constant of valid entity types. Realise that's not completely validating it's a group, but it should protect against anything truly messy.